tensorflow_datasets/core/features/image_feature.py

```diff
@@ -345,3 +370,46 @@ def _get_and_validate_shape(shape, encoding_format):
         f'{acceptable_channels} (was {channels})'
     )
   return tuple(shape)
+
+
+def _get_and_validate_colormap(use_colormap, shape, encoding_format):
+  """Update the shape."""
+  if use_colormap and encoding_format and encoding_format != 'png':
+    raise ValueError(
+        f'Colormap is only available for PNG images. Not {encoding_format}'
+    )
+  if shape[-1] != 1 and use_colormap:
+    raise ValueError('Colormap is only available for gray-scale images')
+
+  return use_colormap
+
+
+@utils.memoize()
+def _get_colormap(cmap_size: int) -> np.ndarray:
+  """Generate cmap configuration"""
+  cmap_list = []
+  cmap_name = 'hsv'
+  matplotlib = lazy_imports_lib.lazy_imports.matplotlib  # pylint: disable=invalid-name
+
+  # cmap returned in the rgba format
+  cmap = matplotlib.cm.get_cmap(cmap_name, cmap_size)
+  for itr in range(cmap_size):
+    cmap_list.append(cmap(itr))
+
+  return cmap_list
+
+
+def _apply_colormap(image: np.ndarray) -> np.ndarray:
+  """Apply colormap to 1D images"""
+  image = image.squeeze(axis=-1)
+  unique_pixels = np.unique(image)
+  colors = _get_colormap(len(unique_pixels))
+  blank_image = np.zeros(shape=(*image.shape[:2], 3), dtype=np.uint8)
+
+  for itr, value in enumerate(unique_pixels):
```

Comment:
It would be much more efficient to use the code snippet I posted in my previous comment:

```python
def _apply_colormap(img: np.ndarray) -> np.ndarray:
  img = img.squeeze(axis=-1)  # (x, y, 1) -> (x, y)
  cmap = _get_colormap()  # (255, 3) id->color mapping 
  return cmap[img]  # Apply mapping: (x, y) -> (x, y, 3)
```

Reply:
This would be if we generate 255 pixels in the colormap, will change if we end up making 255 unique values in the map. ref https://github.com/tensorflow/datasets/pull/2975#discussion_r564447448